WeakAuras/Profiling.lua

```diff
@@ -6,6 +6,132 @@ local profileData = {}
 profileData.systems = {}
 profileData.auras = {}
 
+WeakAuras.table_to_string = function(tbl, depth)
+  if depth and depth >= 3 then
+    return "{ ... }"
+  end
+  local str
+  for k,v in pairs(tbl) do
+    if type(v) ~= "userdata" then
+      if type(v) == "table" then
+        v = WeakAuras.table_to_string(v, (depth and depth + 1 or 1))
+      elseif type(v) == "function" then
+        v = "function"
+      elseif type(v) == "string" then
+        v = '"'.. v ..'"'
+      end
+
+      if type(k) == "string" then
+        k = '"' .. k ..'"'
+      end
+
+      str = (str and str .. "|cff999999,|r " or "|cff999999{|r ") .. "|cffffff99[" .. tostring(k) .. "]|r |cff999999=|r |cffffffff" .. tostring(v) .. "|r"
+    end
+  end
+  return (str or "{ ") .. " }"
+end
+
+local debugpopup
+local function CreateDebugPopup()
+  local p = CreateFrame("EditBox", "WADebugEditBox", UIParent)
+  p:SetFrameStrata("DIALOG")
+  p:SetMultiLine(true)
+  p:SetAutoFocus(true)
+  p:SetFontObject(ChatFontNormal)
+  p:SetSize(450, 300)
+  p:Hide()
+
+  p.orig_Hide = p.Hide
+  function p:Hide()
+    self:SetText("")
+    self.ScrollFrame:Hide()
+    self.Background:Hide()
+    self:orig_Hide()
+  end
+
+  p.orig_Show = p.Show
+  function p:Show()
+    self.ScrollFrame:Show()
+    self.Background:Show()
+    self:orig_Show()
+  end
+
+  function p:AddText(v)
+    if not v then return end
+    local m = p:GetText()
+    if m ~= "" then
+      m = m .. "|n"
+    end
+    if type(v) == "table" then
+      v = WeakAuras.table_to_string(v)
+    end
+    p:SetText(m .. v)
+  end
+
+  p:SetScript("OnEscapePressed", function(self)
+    self:ClearFocus()
+    self:Hide()
+    self.ScrollFrame:Hide()
+    self.Background:Hide()
+  end)
+
+  local s = CreateFrame("ScrollFrame", "WADebugEditBoxScrollFrame", UIParent, "UIPanelScrollFrameTemplate")
+  s:SetMovable(true)
+  s:SetFrameStrata("DIALOG")
+  s:SetSize(450, 300)
+  s:SetPoint("CENTER")
+  s:SetHitRectInsets(-4, -4, -4, -4)
+  s:SetScrollChild(p)
+  s:Hide()
+
+  s:SetScript("OnMouseDown", function(self, button)
+    if button == "RightButton" and not self.is_moving then
```

Comment:
That's rather non-standard for our dialogss.

Reply:
So left click for drag then?

Reply:
Yeah, we have a titlebar for that.